csclient/csclient.go

```diff
@@ -257,28 +256,26 @@ func (s *Client) Publish(id *charm.URL, channels []params.Channel, resources map
 // It must be closed after use.
 type ResourceData struct {
 	io.ReadCloser
-	Revision int
-	Hash     string
-	Size     int64
+	Hash string
+	Size int64
 }
 
 // GetResource retrieves byes of the resource with the given name and revision
 // for the given charm, returning a reader its data can be read from,  the
-// SHA384 hash of the data and its size.  If revision is -1, the latest revision
-// of the resource will be retrieved.
+// SHA384 hash of the data and its size.
 //
 // Note that the result must be closed after use.
-func (c *Client) GetResource(id *charm.URL, revision int, name string) (result ResourceData, err error) {
+func (c *Client) GetResource(id *charm.URL, name string, revision int) (result ResourceData, err error) {
+	if revision < 0 {
+		return result, errgo.New("revision must be a non-negative integer")
+	}
```

Comment:
The revision is optional.  If not provided then the latest is used.  In the client "not provided" is represented by a negative revision.